io/net/htx_nic_devices.py

```diff
@@ -68,22 +68,26 @@ def setUp(self):
             self.cancel("failed connecting to peer")
         self.remotehost = RemoteHost(self.peer_ip, self.peer_user,
                                      password=self.peer_password)
-
         if 'start' in str(self.name.name):
-            # Flush out the ip addresses on host before starting the test
-            for interface in self.host_intfs:
-                cmd = 'ip addr flush dev %s' % interface
-                process.run(cmd, shell=True, sudo=True, ignore_status=True)
-                cmd = 'ip link set dev %s up' % interface
-                process.run(cmd, shell=True, sudo=True, ignore_status=True)
-
-            # Flush out the ip addresses on peer before starting the test
-            for peer_interface in self.peer_intfs:
-                cmd = 'ip addr flush dev %s' % peer_interface
-                self.session.cmd(cmd)
-                cmd = 'ip link set dev %s up' % peer_interface
-                self.session.cmd(cmd)
-
+            for ipaddr, interface in zip(self.host_private_ips, self.host_intfs):
+                if self.hbond:
+                    networkinterface = NetworkInterface(interface, self.localhost, if_type='Bond')
+                else:
+                    networkinterface = NetworkInterface(interface, self.localhost)
+                try:
+                    networkinterface.add_ipaddr(ipaddr, self.netmask)
+                    networkinterface.save(ipaddr, self.netmask)
+                except Exception:
+                    networkinterface.save(ipaddr, self.netmask)
+            for interface in self.peer_intfs:
+                peer_networkinterface = NetworkInterface(interface, self.remotehost)
+            for peerip in self.peer_ipaddrs:
+                try:
+                    peer_networkinterface.add_ipaddr(peerip, self.netmask)
+                    peer_networkinterface.save(peerip, self.netmask)
+                except Exception:
+                    peer_networkinterface.save(peerip, self.netmask)
+        self.ip_addr_remove()
```

Comment:
if you want to flush the IP, why to configure in the first place ?... incase to take a backup of ip configuration.. than every test case does configure network before start...  so just flush and delete the ips and than let htx configure its own ips.... and at the end of htx again.. flush out the thxips  ? am I missing something here ?